### PR TITLE
ci: Added ci package uploads

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -218,7 +218,7 @@ jobs:
         run: python -m pytest tests --cov=armonik --cov-config=.coveragerc --cov-report=term-missing --cov-report xml:coverage.xml --cov-report html:coverage_report
       
       - name: Get Cover 
-        uses: orgoro/coverage@d77626a5fa35d39123e86d6c62907fabe2491496 # v3
+        uses: orgoro/coverage@v3.1
```
Comment (aneojgurhem) [inline .github/workflows/ci.yml:221]: why ? this commit should be the latest 3.x

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -236,9 +236,12 @@ jobs:
           path: packages/python/coverage.xml
 
   build-cpp-packages:
-    name: Build C++ Packages
```
Comment (aneojgurhem) [inline .github/workflows/ci.yml:239]: why remove the name ?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -236,9 +236,12 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        type: [tar.gz, deb, rpm]
```
Comment (aneojgurhem) [inline .github/workflows/ci.yml:242]: I think that a matrix here is not need. You compile 3 times the same thing. Once should be enough

File: .github/workflows/publish-edge.yml [hunk 4]
```diff
@@ -232,3 +232,29 @@ jobs:
           password: ${{ secrets.PYPI_API_TOKEN }}
           print-hash: true
           packages-dir: packages/python/pkg/
+
+    release-cpp-package:
```
Comment (aneojgurhem) [inline .github/workflows/publish-edge.yml:236]: You should have some kind of indentation problem here

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dbrasseur-aneo) [reply .github/workflows/ci.yml:242]: No they are 3 different dockerfiles with different underlying architectures. tar.gz uses ubuntu 23.04, deb uses debian 12 and rpm uses rhel7

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dbrasseur-aneo) [reply .github/workflows/ci.yml:239]: The name is changed a few lines below

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aneojgurhem) [reply .github/workflows/ci.yml:221]: Then put the commit hash associated to 3.1

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**